app/uk/gov/hmrc/incorporatedentityidentificationfrontend/services/RegistrationOrchestrationService.scala

```diff
@@ -27,34 +28,54 @@ import scala.concurrent.{ExecutionContext, Future}
+                                                )(implicit ec: ExecutionContext) extends Logging {
+
+  def register(journeyId: String, journeyConfig: JourneyConfig)(implicit hc: HeaderCarrier): Future[RegistrationStatus] = {
+
+    def shouldRegister(): Future[Boolean] = {
+      storageService.retrieveBusinessVerificationStatus(journeyId).map {
+        case Some(BusinessVerificationPass | CtEnrolled) => true
+        case Some(BusinessVerificationNotEnoughInformationToChallenge |
+                  BusinessVerificationNotEnoughInformationToCallBV |
+                  BusinessVerificationFail) => false
+        case None if !journeyConfig.businessVerificationCheck => true
+        case None =>
+          throw new InternalServerException(s"Missing business verification state in database for $journeyId")
+      }
     }
-    registrationStatus <- if (shouldRegister) for {
-      optCompanyProfile <- storageService.retrieveCompanyProfile(journeyId)
-      optCtutr <- storageService.retrieveCtutr(journeyId)
-      registrationStatus <-
-        (optCompanyProfile, optCtutr) match {
-          case (Some(companyProfile), Some(ctutr)) =>
-            journeyConfig.businessEntity match {
-              case LimitedCompany => registrationConnector.registerLimitedCompany(companyProfile.companyNumber, ctutr, journeyConfig.regime)
-              case RegisteredSociety => registrationConnector.registerRegisteredSociety(companyProfile.companyNumber, ctutr, journeyConfig.regime)
-              case CharitableIncorporatedOrganisation => Future.successful(RegistrationNotCalled) //Not currently registered
-            }
+
+    def register(): Future[RegistrationStatus] = {
+      for {
+        optCompanyProfile <- storageService.retrieveCompanyProfile(journeyId)
+        optCtutr <- storageService.retrieveCtutr(journeyId)
+        registrationStatus <-
+          (optCompanyProfile, optCtutr) match {
+            case (Some(companyProfile), Some(ctutr)) =>
+              journeyConfig.businessEntity match {
+                case LimitedCompany => registrationConnector.registerLimitedCompany(companyProfile.companyNumber, ctutr, journeyConfig.regime)
+                case RegisteredSociety => registrationConnector.registerRegisteredSociety(companyProfile.companyNumber, ctutr, journeyConfig.regime)
+                case CharitableIncorporatedOrganisation => Future.successful(RegistrationNotCalled) //Not currently registered
+              }
+            case _ =>
+              throw new InternalServerException(s"Missing required data for registration in database for $journeyId")
+          }
+      } yield registrationStatus
+    }
+
+    for {
+      sr <- shouldRegister()
+      existingRegistrationStatus <- storageService.retrieveRegistrationStatus(journeyId)
+      registrationStatus <- {
+        existingRegistrationStatus match {
+          case Some(status@Registered(_)) =>
```

Comment:
If we find that a registration has previously been submitted successfully, we don't submit it again and instead return the result of the previous successful submission.